src/main/java/com/jakub/todoSandbox/repository/InMemoryRepository.java

```diff
@@ -0,0 +1,165 @@
+    public List<Todo> findAllTodos() {
+        List<Todo> todoMapToList = new ArrayList<>(todoList.values());
+        return sortByPriority(todoMapToList);
+    }
+
+    @Override
+    public Todo saveTodo(Todo todo) {
+        if (validateNameAndDesc(todo.name(), todo.description())) {
+            var toBeSavedTodo = new Todo((maxTodoId + 1L), todo.name(), todo.description(), todo.priority(), todo.steps());
+            todoList.put(toBeSavedTodo.id(), toBeSavedTodo);
+            maxTodoId = toBeSavedTodo.id();
+            System.out.println(
+                    "New todo was created:" +
+                              "\n id: " + toBeSavedTodo.id()
+                            + " \n name: " + toBeSavedTodo.name()
+                            + "\n description: " + toBeSavedTodo.description()
+                            + "\n priority: " + toBeSavedTodo.priority()
+                            + "\n steps: \n" + toBeSavedTodo.steps().get(0).name());
+            return toBeSavedTodo;
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public void updateTodo(Long todoId, Todo todo) {
+        var exists = todoList.get(todoId);
+        if (exists != null) {
+            todoList.put(todoId, createNewTodoId(todoId, todo));
+            System.out.println("Todo using id: {" + todoId + "} has been updated.");
+        } else {
+            System.out.println("No todo exists under this id.");
+        }
+    }
+
+    @Override
+    public void deleteTodo(Long todoId) {
+        if (todoList.get(todoId) != null) {
+            todoList.remove(todoId);
+            System.out.println("Todo with the id: " + todoId + " was removed.");
+        } else {
+            System.out.println("There isn't any todo under id: " + todoId);
+        }
+    }
+
+    @Override
+    public void saveSteps(Long todoId, List<Step> createdSteps) {
+        var existingTodo = todoList.get(todoId);
+        if (existingTodo != null) {
+            System.out.println("Size of step array before adding new: " + existingTodo.steps().size());
+            for (Step step : createdSteps) {
+                if (canAddStepToTodo(existingTodo)) {
+                    Step createdStep = createNewStepForTodo(existingTodo, step);
+                    existingTodo.steps().add(createdStep);
+                } else {
+                    System.out.println("You have reached the maximum number of steps of 10 for todo with id: " + existingTodo.id());
+                    break;
+                }
+            }
+            System.out.println("Size of step array after adding new: " + existingTodo.steps().size());
+        } else {
+            System.out.println("Such a Todo does not exist.");
+        }
+    }
+
+    @Override
+    public void updateStep(Long todoId, int oldStepId, Step updatedStep) {
+        var existingTodo = todoList.get(todoId);
+        if (existingTodo != null) {
+            for (int i = 0; i < existingTodo.steps().size(); i++) {
```

Comment:
1. Validation should be at the higher level
2. We should throw exception any case of failed validation